wondrous-app/services/snapshot/index.tsx

```diff
@@ -0,0 +1,362 @@
+        } else {
+          setSnapshotValid(false);
+          handleOnboardingError(data.space.id, Onboarding.NotOwner);
+        }
+      } else {
+        setSnapshotValid(false);
+        handleOnboardingError(snapshotName, Onboarding.NotFound);
+      }
+    },
+    onError: error => {
+      console.error(`Error validateSnapshotSpace(): ${error}`);
+    },
+    fetchPolicy: 'cache-and-network'
+  });
+
+  // gets Snapshot via Wonder API, doesn't update local state
+  const [getSnapshot, { loading: getSnapshotLoading }] = useLazyQuery(GET_SNAPSHOT, {
+    onError: error => {
+      console.error(`Error getSnapshot(): ${error}`);
+    },
+    fetchPolicy: 'network-only'
+  });
+
+  // checks for Snapshot stored in Wonder Org via Wonder API & updates state to reflect db
+  const [validateSnapshot, { loading: validateSnapshotLoading }] = useLazyQuery(GET_SNAPSHOT, {
+    onCompleted: data => {
+      // returns 'null' if there no data is in the integrations column of 'org' table
+      const snapshot = data.getOrgById.integrations
+        ? data.getOrgById.integrations[0]
+        : null;
+      // sets local state of integrations field from 'org' table, or null if nonexistent
+      setSnapshot(snapshot);
+      // sets cached name, to ensure persistence
+      setSnapshotName(snapshot.key);
+      // updates snapshot connection state
+      setSnapshotConnected(snapshot.key ? true : false);
+    },
+    onError: error => {
+      console.error(`Error validateSnapshot(): ${error}`);
+    },
+    fetchPolicy: 'network-only'
+  });
+
+  // updates Wonder db & local state w/ snapshot integrations data
+  const [connectSnapshotSpace, { loading: connectSnapshotSpaceLoading }] = useMutation(SET_SPACE, {
+    onCompleted: data => {
+      const [integrations] = data.updateOrg.integrations
+      setSnapshot(integrations)
+      setSnapshotConnected(true);
+    },
+    onError: error => {
+      console.error(`Error connectSnapshot(): ${error}`);
+    },
+  });
+
+  // disconnects snapshot space in Wonder db & local state
+  // (same as connectSnapshotSpace, except updates snapshotConnection to false)
+  const [disconnectSnapshotSpace, { loading: disconnectSnapshotSpaceLoading }] = useMutation(SET_SPACE, {
+    onCompleted: data => {
+      const [integrations] = data.updateOrg.integrations
+      setSnapshot(integrations)
+      setSnapshotConnected(false);
+    },
+    onError: error => {
+      console.error(`Error disconnectSnapshot(): ${error}`);
+    },
+  });
+
+  // update Task Proposal's snapshot_proposal field
+  const [updateSnapshotProposal, { loading: updateSnapshotProposalLoading }] = useMutation(SET_SNAPSHOT_PROPOSAL, {
+    onError: error => {
+      console.error(`error updateSnapshotProposal(): ${error}`);
+    }
+  });
+
+  // creates transaction to export proposal to snapshot, throws error if exporter doesn't have credentials or invalid proposal
+  // HAVING ISSUES, trying to solve w/ Snapshot team
```

Comment:
Is this still a problem?

Reply:
Forgot to get rid of that comment, that was for the snippets I provided to snapshot